app.go

```diff
@@ -674,8 +675,13 @@ func (app *App) run(done <-chan os.Signal) (exitCode int) {
 	startCtx, cancel := context.WithTimeout(context.Background(), app.StartTimeout())
 	defer cancel()
 
+	exitCode = 1
+	if app.exitCode != 0 {
+		exitCode = app.exitCode
```

Comment:
This won't give us what we want for normal use cases.
Per our documentation, Shutdowner should be invoked from a lifecycle hook.
That's run when we call App.Start.
By that point, we'll have already captured the exit code.

We need to use the new DoneWithCode API below after the Start, and use the exit code from that.